acvm-repo/brillig_vm/src/black_box.rs

```diff
@@ -320,39 +319,112 @@ pub(crate) fn evaluate_black_box<F: AcirField, Solver: BlackBoxFunctionSolver<F>
                 panic!("ToRadix opcode's output_bits size does not match expected bit size 1")
             };
 
-            let mut input = BigUint::from_bytes_be(&input.to_be_bytes());
-            let radix = BigUint::from_bytes_be(&radix.to_be_bytes());
-
-            let mut limbs: Vec<MemoryValue<F>> = vec![MemoryValue::default(); num_limbs];
-
             assert!(
-                radix >= BigUint::from(2u32) && radix <= BigUint::from(256u32),
+                (2..=256).contains(&radix),
                 "Radix out of the valid range [2,256]. Value: {}",
                 radix
             );
 
             assert!(
-                num_limbs >= 1 || input == BigUint::from(0u32),
+                num_limbs >= 1 || input.is_zero(),
                 "Input value {} is not zero but number of limbs is zero.",
                 input
             );
 
             assert!(
-                !output_bits || radix == BigUint::from(2u32),
+                !output_bits || radix == 2,
                 "Radix {} is not equal to 2 and bit mode is activated.",
                 radix
             );
 
-            for i in (0..num_limbs).rev() {
-                let limb = &input % &radix;
-                if output_bits {
-                    limbs[i] = MemoryValue::U1(!limb.is_zero());
-                } else {
-                    let limb: u8 = limb.try_into().unwrap();
-                    limbs[i] = MemoryValue::U8(limb);
-                };
-                input /= &radix;
-            }
+            // TODO: raise an error if we cannot fit `input` into `num_limbs`.
+            let limbs = match radix {
+                256 => input
+                    .to_be_bytes()
+                    .into_iter()
+                    .rev()
+                    .take(num_limbs)
+                    .rev()
+                    .map(|limb| MemoryValue::U8(limb))
+                    .collect(),
+                2 => input
+                    .to_be_bytes()
+                    .into_iter()
+                    .rev()
+                    .take(num_limbs.div_ceil(8))
```

Comment:
Can you fix the formatting on these call chains? As-is I think it makes it more difficult to compare where the different cases differ since they're stretched across more lines.

Possibly refactoring out repeated portions as well. E.g. it seems at least the following portion is repeated:

```rs
to_be_bytes()
    .into_iter()
    .rev()
    .take(arg_here_differs)
    .rev()
```